File: ios-client/StripeApplication/StripeApplication/Shared/Views/AccountCard.swift
import SwiftUI

struct AccountCard: View {
    let account: Account
    let isLoading: Bool
    let onDelete: () -> Void
    let onUpgradeToRecipient: () -> Void
    let onStartOnboarding: () -> Void

    @State private var showDeleteDialog = false

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            // Header
            HStack {
                Text("Account Details")
                    .font(.headline)
                Spacer()
                Button(role: .destructive) {
                    showDeleteDialog = true
                } label: {
                    Image(systemName: "trash")
                        .foregroundStyle(.red)
                }
                .disabled(isLoading)
            }

            // Status badges
            HStack(spacing: 8) {
                if account.isCustomer {
                    BadgeView(text: "Customer", color: .secondary)
                }
                if account.isRecipient {
                    BadgeView(text: "Recipient", color: .blue)
                }
                if account.isOnboarding == true {
                    BadgeView(text: "Onboarding", color: .orange)
                }
            }

            Divider()

            // Account info
            InfoRow(label: "Email", value: account.email ?? "N/A")
            InfoRow(label: "Display Name", value: account.displayName ?? "N/A")
            InfoRow(label: "Platform ID", value: account.id)
            InfoRow(label: "Stripe Account ID", value: account.stripeAccountId)
            if let customerId = account.stripeCustomerId {
                InfoRow(label: "Stripe Customer ID", value: customerId)
            }
            InfoRow(label: "Created", value: account.created)

            // Actions
            if !account.isRecipient {
                Divider()
                Button {
                    onUpgradeToRecipient()
                } label: {
                    HStack {
                        Spacer()
                        if isLoading {
                            ProgressView()
                                .tint(.white)
                        } else {
                            Text("Upgrade to Recipient")
                        }
                        Spacer()
                    }
                }
                .buttonStyle(.borderedProminent)
                .disabled(isLoading)
            }

            if account.isRecipient && account.isOnboarding == true {
                Divider()
                Button {
                    onStartOnboarding()
                } label: {
                    HStack {
                        Spacer()
                        Text("Complete Onboarding")
                        Spacer()
                    }
                }
                .buttonStyle(.bordered)
                .disabled(isLoading)
            }
        }
        .padding()
        .background(
            RoundedRectangle(cornerRadius: 12)
                .fill(Color(.systemBackground))
                .shadow(color: .black.opacity(0.1), radius: 4, y: 2)
        )
        .alert("Delete Account", isPresented: $showDeleteDialog) {
            Button("Cancel", role: .cancel) {}
            Button("Delete", role: .destructive) {
                onDelete()
            }
        } message: {
            Text("Are you sure you want to delete this account? This action cannot be undone.")
        }
    }
}

struct InfoRow: View {
    let label: String
    let value: String

    var body: some View {
        HStack {
            Text(label)
                .foregroundStyle(.secondary)
                .font(.subheadline)
            Spacer()
            Text(value)
                .font(.subheadline)
                .lineLimit(1)
                .truncationMode(.middle)
        }
    }
}

struct BadgeView: View {
    let text: String
    let color: Color

    var body: some View {
        Text(text)
            .font(.caption)
            .fontWeight(.medium)
            .padding(.horizontal, 10)
            .padding(.vertical, 4)
            .background(color.opacity(0.15))
            .foregroundStyle(color)
            .clipShape(Capsule())
    }
}

File: android-client/app/src/main/java/com/example/stripedemo/features/accountdetail/components/AccountCard.kt
package com.example.stripedemo.features.accountdetail.components

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.example.stripedemo.data.models.Account

@Composable
fun AccountCard(
    account: Account,
    isLoading: Boolean,
    onDelete: () -> Unit,
    onUpgradeToRecipient: () -> Unit,
    onStartOnboarding: () -> Unit,
    modifier: Modifier = Modifier
) {
    var showDeleteDialog by remember { mutableStateOf(false) }

    Card(
        modifier = modifier.fillMaxWidth()
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Account Details",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold
                )

                IconButton(
                    onClick = { showDeleteDialog = true },
                    enabled = !isLoading
                ) {
                    Icon(
                        imageVector = Icons.Default.Delete,
                        contentDescription = "Delete Account",
                        tint = MaterialTheme.colorScheme.error
                    )
                }
            }

            // Status badges
            Row(
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                if (account.isCustomer) {
                    SuggestionChip(
                        onClick = {},
                        label = { Text("Customer") }
                    )
                }
                if (account.isRecipient) {
                    SuggestionChip(
                        onClick = {},
                        label = { Text("Recipient") },
                        colors = SuggestionChipDefaults.suggestionChipColors(
                            containerColor = MaterialTheme.colorScheme.primaryContainer
                        )
                    )
                }
                if (account.isOnboarding == true) {
                    SuggestionChip(
                        onClick = {},
                        label = { Text("Onboarding") },
                        colors = SuggestionChipDefaults.suggestionChipColors(
                            containerColor = MaterialTheme.colorScheme.tertiaryContainer
                        )
                    )
                }
            }

            HorizontalDivider()

            // Account info
            AccountInfoRow("Email", account.email ?: "N/A")
            AccountInfoRow("Display Name", account.displayName ?: "N/A")
            AccountInfoRow("Platform ID", account.id)
            AccountInfoRow("Stripe Account ID", account.stripeAccountId)
            account.stripeCustomerId?.let {
                AccountInfoRow("Stripe Customer ID", it)
            }
            AccountInfoRow("Created", account.created)

            // Actions
            if (!account.isRecipient) {
                HorizontalDivider()
                Button(
                    onClick = onUpgradeToRecipient,
                    enabled = !isLoading,
                    modifier = Modifier.fillMaxWidth()
                ) {
                    if (isLoading) {
                        CircularProgressIndicator(
                            modifier = Modifier.size(20.dp),
                            color = MaterialTheme.colorScheme.onPrimary,
                            strokeWidth = 2.dp
                        )
                    } else {
                        Text("Upgrade to Recipient")
                    }
                }
            }

            if (account.isRecipient && account.isOnboarding == true) {
                HorizontalDivider()
                OutlinedButton(
                    onClick = onStartOnboarding,
                    enabled = !isLoading,
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Text("Complete Onboarding")
                }
            }
        }
    }

    if (showDeleteDialog) {
        AlertDialog(
            onDismissRequest = { showDeleteDialog = false },
            title = { Text("Delete Account") },
            text = { Text("Are you sure you want to delete this account? This action cannot be undone.") },
            confirmButton = {
                TextButton(
                    onClick = {
                        showDeleteDialog = false
                        onDelete()
                    },
                    colors = ButtonDefaults.textButtonColors(
                        contentColor = MaterialTheme.colorScheme.error
                    )
                ) {
                    Text("Delete")
                }
            },
            dismissButton = {
                TextButton(onClick = { showDeleteDialog = false }) {
                    Text("Cancel")
                }
            }
        )
    }
}

@Composable
private fun AccountInfoRow(
    label: String,
    value: String,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Text(
            text = label,
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
        Text(
            text = value,
            style = MaterialTheme.typography.bodyMedium
        )
    }
}
